Recover updates to docs and release notes (#356)
kubernetes-sigs/kueue: docs/concepts/cluster_queue.md

## Resources

In a ClusterQueue, you can define quotas for multiple [compute resources](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#resource-types)
(CPU, memory, GPUs, etc.).

For each resource, you can define quotas for multiple _flavors_.
Flavors represent different variations of a resource (for example, different GPU

> models). A flavor can be defined using a [ResourceFlavor object](#resourceflavor-object).

Reviewer: ```suggestion
models). A flavor is defined using a [ResourceFlavor object](#resourceflavor-object).
```

Author: Done

In a process called [admission](.#admission), Kueue assigns to the
[Workload pod sets](workload.md#pod-sets) a flavor for each resource the pod set
requests.
Kueue assigns the first flavor in the ClusterQueue's `.spec.resources[*].flavors`
list that has enough unused `min` quota in the ClusterQueue or the
ClusterQueue's [cohort](#cohort).

### Codepedent resources

It is possible that multiple resources in a ClusterQueue have the same flavors.
This is typical for `cpu` and `memory`, where the flavors are generally tied to
a machine family or VM availability policies. When two or more resources in a
ClusterQueue match their flavors, they are said to be codependent resources.

> To manage codependent resources, list the flavors in the ClusterQueue resources

Reviewer: ```suggestion
To manage codependent resources, you should list the flavors in the ClusterQueue resources
```

Author: Done

### Flavors and borrowing semantics

When a ClusterQueue is part of a cohort, Kueue satisfies the following admission
semantics:

- When assigning flavors, Kueue goes through the list of flavors in the
  ClusterQueue's `.spec.resources[*].flavors`. For each flavor, Kueue attempts
  to fit a Workload's pod set according to the quota defined in the
  ClusterQueue for the flavor and the unused quota in the cohort.

>   If the workload doesn't fit, Kueue proceeds evaluating the next flavor in the

Reviewer: ```suggestion
  If the workload doesn't fit, Kueue evaluates the next flavor in the
```

Author: Done
